src/checkers/inference/InferenceLauncher.java

```diff
@@ -0,0 +1,355 @@
+        new InferenceLauncher(System.out, System.err).launch(args);
+    }
+
+    public void typecheck(String [] javaFiles) {
+        printStep("Typechecking", outStream);
+
+        final int initialOptsLength = 2 + (InferenceOptions.debug != null ? 2 : 0);
+
+        String [] options;
+        options = new String[initialOptsLength + InferenceOptions.javacOptions.length + javaFiles.length];
+        options[0] = "-processor";
+        options[1] = InferenceOptions.checker;
+
+        if (InferenceOptions.debug != null) {
+            options[2] = "-J-Xdebug";
+            options[3] = "-J-Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=" + InferenceOptions.debug;
+        }
+
+        System.arraycopy(InferenceOptions.javacOptions, 0, options, initialOptsLength, InferenceOptions.javacOptions.length);
+        System.arraycopy(javaFiles, 0, options, InferenceOptions.javacOptions.length + initialOptsLength, javaFiles.length);
+
+        final CheckerMain checkerMain = new CheckerMain(InferenceOptions.checkerJar, options);
+        checkerMain.addToRuntimeBootclasspath(getInferenceRuntimeBootJars());
+
+        if (InferenceOptions.printCommands) {
+            outStream.println("Running typecheck command:");
+            outStream.println(PluginUtil.join(" ", checkerMain.getExecArguments()));
+        }
+
+        int result = checkerMain.invokeCompiler();
+
+        reportStatus("Typechecking", result, outStream);
+        outStream.flush();
+        exitOnNonZeroStatus(result);
+    }
+
+    public void infer() {
+        printStep("Inferring", outStream);
+        final String java = PluginUtil.getJavaCommand(System.getProperty("java.home"), outStream);
+        List<String> argList = new LinkedList<>();
+        argList.add(java);
+        argList.addAll(getMemoryArgs());
+
+        if (InferenceOptions.debug != null) {
+            argList.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=" + InferenceOptions.debug);
+        }
+
+        argList.add(getInferenceRuntimeBootclassPath());
+        argList.addAll(
+                Arrays.asList(
+                        "-ea", "-ea:checkers.inference...",
+                        "checkers.inference.InferenceMain",
+                        "--checker", InferenceOptions.checker)
+        );
+
+        addIfNotNull("--jaifFile", InferenceOptions.jaifFile, argList);
+        addIfNotNull("--logLevel", InferenceOptions.logLevel, argList);
+        addIfNotNull("--solver", InferenceOptions.solver, argList);
+        addIfNotNull("--solverArgs", InferenceOptions.solverArgs, argList);
+
+        addIfTrue("--hacks", InferenceOptions.hacks, argList);
+
+        argList.add("--");
+        argList.add(getInferenceCompilationBootclassPath());
+        int preJavacOptsSize = argList.size();
+        argList.addAll(Arrays.asList(InferenceOptions.javacOptions));
+        removeXmArgs(argList, preJavacOptsSize, argList.size());
+
+        //TODO: NEED TO HANDLE JDK
```

Comment:
THis TODO will be removed.  The JDK is added via getInfenceCompilationBootclasspath.